src/Domain/TagsTimeDependent.hpp

```diff
@@ -0,0 +1,153 @@
+          functions_of_time) noexcept {
+    // Use identity to signal time-independent
+    if (not grid_to_inertial_map.is_identity()) {
+      *result = grid_to_inertial_map.coords_frame_velocity_jacobians(
+          source_coords, time, functions_of_time);
+    } else {
+      *result = boost::none;
+    }
+  }
+
+  using argument_tags =
+      tmpl::list<MapTagGridToInertial, Tags::Coordinates<dim, Frame::Grid>,
+                 ::Tags::Time, Tags::FunctionsOfTime>;
+};
+
+/// Computes the Inertial coordinates from
+/// `CoordinatesVelocityAndJacobians`
+template <size_t Dim>
+struct InertialCoordinates : Tags::Coordinates<Dim, Frame::Inertial>,
+                             db::ComputeTag {
+  using base = Tags::Coordinates<Dim, Frame::Inertial>;
+  using return_type = typename base::type;
+
+  static void function(
+      gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> target_coords,
+      const tnsr::I<DataVector, Dim, Frame::Grid>& source_coords,
+      const boost::optional<std::tuple<
+          tnsr::I<DataVector, Dim, Frame::Inertial>,
+          ::InverseJacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+          ::Jacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+          tnsr::I<DataVector, Dim, Frame::Inertial>>>&
+          grid_to_inertial_quantities) noexcept;
+
+  using argument_tags = tmpl::list<Tags::Coordinates<Dim, Frame::Grid>,
+                                   CoordinatesVelocityAndJacobians<Dim>>;
+};
+
+/// Computes the Logical to Inertial inverse Jacobian from
+/// `CoordinatesVelocityAndJacobians`
+template <size_t Dim>
+struct ElementToInertialInverseJacobian
+    : Tags::InverseJacobian<Dim, Frame::Logical, Frame::Inertial>,
+      db::ComputeTag {
+  using base = Tags::InverseJacobian<Dim, Frame::Logical, Frame::Inertial>;
+  using return_type = typename base::type;
+
+  static void function(
+      gsl::not_null<
+          ::InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+          inv_jac_logical_to_inertial,
+      const ::InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Grid>&
+          inv_jac_logical_to_grid,
+      const boost::optional<std::tuple<
+          tnsr::I<DataVector, Dim, Frame::Inertial>,
+          ::InverseJacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+          ::Jacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+          tnsr::I<DataVector, Dim, Frame::Inertial>>>&
+          grid_to_inertial_quantities) noexcept;
+
+  using argument_tags =
+      tmpl::list<Tags::InverseJacobian<Dim, Frame::Logical, Frame::Grid>,
+                 CoordinatesVelocityAndJacobians<Dim>>;
+};
+
+/// The frame velocity
+///
+/// The type is a `boost::optional`, which when it is not set indicates that the
+/// mesh is not moving.
+template <size_t Dim, typename Frame = ::Frame::Inertial>
+struct FrameVelocity : db::SimpleTag {
+  using type = boost::optional<tnsr::I<DataVector, Dim, Frame>>;
```

Comment:
I'm confused why this is a `tnsr::I`.. shouldn't it be a `tnsr::i`?

Reply:
because the coordinates are `x^i` and not `x_i`... https://arxiv.org/pdf/1510.01190.pdf the section on moving meshes might be helpful